src/lib/comms/sol-oic-client.c

```diff
@@ -1045,11 +1076,11 @@ _observe_with_polling(struct sol_oic_client *client, struct sol_oic_resource *re
     void *data)
 {
     struct resource_request_ctx *ctx = sol_util_memdup(&(struct resource_request_ctx) {
```

Comment:
We are using this context in a sol_timeout, so we need to allocated memory for this.